node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorDatabaseClient.java

```diff
@@ -473,23 +473,28 @@ private Path firmwareCheckPath() {
 
 
     // Load balancers
-    public Map<LoadBalancerId, LoadBalancer> readLoadBalancers() {
+    public List<LoadBalancerId> readLoadBalancerIds() {
         return curatorDatabase.getChildren(loadBalancersRoot).stream()
                               .map(LoadBalancerId::fromSerializedForm)
-                              .map(this::readLoadBalancer)
-                              .filter(Optional::isPresent)
-                              .map(Optional::get)
-                              .collect(collectingAndThen(toMap(LoadBalancer::id, Function.identity()),
-                                                         Collections::unmodifiableMap));
+                              .collect(Collectors.toUnmodifiableList());
+    }
+
+    public Map<LoadBalancerId, LoadBalancer> readLoadBalancers() {
+        return readLoadBalancerIds().stream()
+                                    .map(this::readLoadBalancer)
+                                    .filter(Optional::isPresent)
+                                    .map(Optional::get)
+                                    .collect(collectingAndThen(toMap(LoadBalancer::id, Function.identity()),
```

Comment:
`Collectors.toUnmodifiableMap()`?

Reply:
It already does. Line below 🙂

Reply:
Ah, it can be simplified now. Thanks!

Reply:
`Collectors.toUnmodfiableMap` results in a different iteration order compared to `collectingAndThen(toMap, unmodifiableMap)` which failed a test so I left it like this.